src/base/bittorrent/session.cpp

```diff
@@ -3441,17 +3446,20 @@ void Session::handleTorrentRemovedAlert(libt::torrent_removed_alert *p)
 
 void Session::handleTorrentDeletedAlert(libt::torrent_deleted_alert *p)
 {
-    m_savePathsToRemove.remove(p->info_hash);
+    const QString path = m_savePathsToRemove.take(p->info_hash);
+    if (path == torrentTempPath(p->info_hash))
+        Utils::Fs::removeDirRecursive(path);
 }
 
 void Session::handleTorrentDeleteFailedAlert(libt::torrent_delete_failed_alert *p)
 {
     // libtorrent won't delete the directory if it contains files not listed in the torrent,
     // so we remove the directory ourselves
-    if (m_savePathsToRemove.contains(p->info_hash)) {
-        QString path = m_savePathsToRemove.take(p->info_hash);
+    const QString path = m_savePathsToRemove.take(p->info_hash);
+    if (path == torrentTempPath(p->info_hash))
+        Utils::Fs::removeDirRecursive(path);
```

Comment:
maybe best handled by `smartRemoveEmptyFolderTree` here.
that said, then this function go back to original form.